runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h

```diff
@@ -43,50 +50,170 @@ void configureHeader();
 void insertNoOpCommand(unsigned int numPadBytes);
 }
 
-#define OSTREAM_OP(O_TYPE, TYPE) O_TYPE &operator<<(O_TYPE &os, const TYPE &s);
+#define XAIE2_BASE_ADDR 0x40000000
+#define XAIE1_BASE_ADDR 0x20000000000
+#define XAIE1_COL_SHIFT 23
+#define XAIE1_ROW_SHIFT 18
+#define XAIE2_BASE_ADDR 0x40000000
+#define XAIE2_COL_SHIFT 25
+#define XAIE2_ROW_SHIFT 20
+#define XAIE_MEM_TILE_ROW_START 1
+#define XAIE_NUM_MEM_TILE_ROWS 1
+#define XAIE_NUM_SHIM_TILE_ROWS 1
+#define XAIE_PARTITION_BASE_ADDR 0x0
+#define XAIE_SHIM_ROW 0
+
+#define NPI_ADDR 0x0
+#define NUM_LOCKS 16
+#define MEM_TILE_LOCK_ID_INCR 64
+#define BASE_ADDR_A_INCR 0x80000
 
 namespace mlir::iree_compiler::AMDAIE {
-#define TO_STRING(TYPE) std::string to_string(const TYPE &t);
+struct TileLoc {
+  int col, row;
+  inline bool operator<(const TileLoc& rhs) const {
+    return std::tie(col, row) < std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator==(const TileLoc& rhs) const {
+    return std::tie(col, row) == std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator!=(const TileLoc& rhs) const { return !(*this == rhs); }
+
+  operator XAie_LocType() const { return XAie_TileLoc(col, row); }
+  TileLoc(XAie_LocType loc) : col(loc.Col), row(loc.Row) {}
+  TileLoc(int col, int row) : col(col), row(row) {}
+};
+
+enum class AMDAIEDevice : uint32_t {
+  xcvc1902 = 1,
+  xcve2302 = 2,
+  xcve2802 = 3,
+  npu = 4,
+  npu1 = 4,
+  npu1_1col = 5,
+  npu1_2col = 6,
+  npu1_3col = 7,
+  npu1_4col = 8,
+};
+
+enum class AMDAIETileType : uint8_t {
+  AIETILE = 0U,
+  SHIMNOC = 1U,
+  SHIMPL = 2U,
+  MEMTILE = 3U,
+  MAX = 4U
+};
+
+struct AMDAIEDeviceModel {
```

Comment:
I think this struct could use some documentation.

Reply:
Definitely needs documentation! Also, where is the class implemented?